core/applyHooks.ts

```diff
@@ -26,7 +32,7 @@ function dispatch(
   event: string,
   store: Store<any>,
   trans: Transition
-): TransitionHookFn {
+): () => void {
```

Comment:
This method returns a function that takes in no params, so it's not really a `TransitionHookFn`. It could be the case the the transition that gets passed into this return function is the actual `trans` value to use rather than the one passed into the higher-order function, but I didn't feel like investigating that.

Reply:
Good point, I agree `() => void` is better type here